script.js

```diff
@@ -1,121 +1,187 @@
-  lebelForSearch.textContent = "Search";
+  const countElementItem = document.createElement("p");
+  countElementItem.id = "count-element-item";
+  // countElementItem.textContent = `${allEpisodes.length}/${allEpisodes.length}`;
 
-  const searchInput = document.createElement("input");
-  searchInput.id = "search-input";
+  countContainer.appendChild(countElementItem);
 
-  const countElementContainer = document.createElement("div");
-  const countElementItem = document.createElement("p");
+  return countContainer;
+}
 
+function populateCountFunction(allEpisodes) {
+  const countElementItem = document.getElementById("count-element-item");
   countElementItem.textContent = `${allEpisodes.length}/${allEpisodes.length}`;
+}
+
+function searchFunction(allEpisodes) {
+  const searchContainer = document.createElement("div");
+  searchContainer.className = "search-and-count-container";
 
-  countElementContainer.appendChild(countElementItem);
-  searcAndCountContainer.appendChild(countElementContainer);
+  const labelForSearch = document.createElement("label");
+  labelForSearch.for = "search-input";
+  labelForSearch.className = "search-lebel";
+  labelForSearch.textContent = "Search";
+
+  const searchInput = document.createElement("input");
+  searchInput.id = "search-input";
 
+  searchContainer.appendChild(searchInput);
   searchInput.addEventListener("input", (e) => {
     const inputResult = e.target.value.trim();
     console.log(inputResult);
 
     const searchResult = allEpisodes.filter(
-      (key) => 
-      key.name.toLowerCase().includes(inputResult.toLowerCase()) ||
-      key.summary.toLowerCase().includes(inputResult.toLowerCase())
-    )
-    makePageForEpisodes(searchResult);
+      (key) =>
+        key.name.toLowerCase().includes(inputResult.toLowerCase()) ||
+        key.summary.toLowerCase().includes(inputResult.toLowerCase())
+    );
 
-    countElementItem.textContent = `${searchResult.length}/${allEpisodes.length}`;
-  })
-  searcAndCountContainer.appendChild(searchInput);
+    makePageForEpisodes(searchResult);
 
-  return searcAndCountContainer
+    buildCountFunction(searchInput);
+    console.log(searchResult, allEpisodes);
+  });
+  return searchContainer;
 }
 
-
-function makeHeader(allEpisodes){
+function makeHeader(allEpisodes) {
   const header = document.createElement("header");
   header.className = "header-container";
 
   const title = document.createElement("h1");
   title.className = "headers-title";
   title.textContent = "TV-DOM-PROJECT";
 
-  const selectEpisodeMainContainer = document.createElement("div");
-  const selectEpisodeContainer = episodesSearchBarFunction(allEpisodes);
-  selectEpisodeMainContainer.appendChild(selectEpisodeContainer);
-  
-  const searcAndCountMainContainer = document.createElement("div");
-  const searcAndCountContainer = searcAndCountFunction(allEpisodes);
-  searcAndCountMainContainer.appendChild(searcAndCountContainer);
-  
+  const builtShowSelect = buildShowSelect();
```

Comment:
nit: just name the variable showsSelect instead of builtShowSelect. Similar for episodesSelect.